main.go

```diff
@@ -1,43 +1,77 @@
 package main
 
 import (
+	"errors"
 	"log"
 	"net/http"
+	"net/http/httputil"
+	"net/url"
 	"observability-remote-write-proxy/pkg/remotewrite"
+	"os"
+)
+
+const (
+	prefixHeader = "X-Forwarded-Prefix"
+)
+
+var (
+	PORT     = os.Getenv("PORT")
+	upstream = os.Getenv("UPSTREAM")
 )
 
 func main() {
-	handler := http.HandlerFunc(handleRequestSuccess)
-	http.Handle("/", handler)
-	err := http.ListenAndServe(":8080", nil)
+	upstreamUrl, err := url.Parse(upstream)
+	if err != nil {
+		panic(err)
+	}
+
+	proxy := httputil.ReverseProxy{
+		Director: func(request *http.Request) {
+			request.URL.Scheme = upstreamUrl.Scheme
+			request.Host = upstreamUrl.Host
+			request.URL.Host = upstreamUrl.Host
+			request.URL.Path = upstreamUrl.Path
+			request.Header.Add(prefixHeader, "/")
+		},
+	}
+
+	proxy.Transport = http.DefaultTransport
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		err := validateRequest(w, r)
```

Comment:
I'm thinking, `validateRequest` should not handle the response, instead just look at the request and return an error or nil.

Reply:
Because we're now setting the status in different places: in `validateRequest` and also in this handler function.

Reply:
thanks @pb82 . just changed it